src/QsCompiler/Compiler/ExternalRewriteSteps.cs

```diff
@@ -17,50 +17,88 @@ namespace Microsoft.Quantum.QsCompiler
 {
     internal static class RewriteSteps
     {
-        /// <summary>
-        /// Concrete implementation of the rewrite steps options such that the configured options may be null. 
-        /// </summary>
-        internal class RewriteStepOptions : IRewriteStepOptions
-        {
-            public readonly string OutputFolder;
-            internal RewriteStepOptions(CompilationLoader.Configuration config) =>
-                this.OutputFolder = config.BuildOutputFolder;
-        }
-
         /// <summary>
         /// Concrete implementation of a rewrite steps with an additional property specifying the dll it was loaded from. 
         /// </summary>
         internal class LoadedStep : IRewriteStep
         {
-            private readonly IRewriteStep Implementation;
             internal readonly Uri Origin;
+            private string _GetOutputFolder;
+            private readonly Action<string> _SetOutputFolder;
+
+            private readonly Func<QsCompilation, (bool, QsCompilation)> _Transformation;
+            private readonly Func<QsCompilation, bool> _PreconditionVerification;
+            private readonly Func<QsCompilation, bool> _PostconditionVerification;
 
             internal LoadedStep(IRewriteStep step, Uri origin)
             {
-                this.Implementation = step ?? throw new ArgumentNullException(nameof(step));
-                this.Origin = origin ?? throw new ArgumentNullException(nameof(origin)); 
+                this.Origin = origin ?? throw new ArgumentNullException(nameof(origin));
+                this.Priority = step?.Priority ?? throw new ArgumentNullException(nameof(step));
+                this.Name = step.Name;
+                this._GetOutputFolder = step.OutputFolder;
+                this._SetOutputFolder = value => step.OutputFolder = value;
```

Comment:
Why is the OutputFolder treated differently then the other properties? Why can't you do this.OutputFolder = step.OutputFolder?